test/shopify-cli/theme/dev_server/base_proxy_test.rb

```diff
@@ -0,0 +1,264 @@
+# frozen_string_literal: true
+require "test_helper"
+require "shopify_cli/theme/dev_server/base_proxy"
+require "rack/mock"
+require "timecop"
+
+module ShopifyCLI
+  module Theme
+    module DevServer
+      class BaseProxyTest < Minitest::Test
+        SECURE_SESSION_ID = "deadbeef"
+
+        def setup
+          super
+          root = ShopifyCLI::ROOT + "/test/fixtures/theme"
+          @ctx = TestHelpers::FakeContext.new(root: root)
+          @proxy = FakeBaseProxy.new(@ctx)
+        end
+
+        def test_get_is_proxied_to_online_store
+          stub_request(:get, "https://dev-theme-server-store.myshopify.com/?_fd=0&pb=0")
+            .with(
+              body: nil,
+              headers: default_proxy_headers,
+            )
+            .to_return(status: 200)
+
+          stub_session_id_request
+
+          request.get("/")
+        end
+
+        def test_refreshes_session_cookie_on_expiry
+          stub_request(:get, "https://dev-theme-server-store.myshopify.com/?_fd=0&pb=0")
+            .with(
+              body: nil,
+              headers: default_proxy_headers,
+            )
+            .to_return(status: 200)
+            .times(2)
+
+          stub_session_id_request
+          request.get("/")
+
+          # Should refresh the session cookie after 1 day
+          Timecop.freeze(DateTime.now + 1) do # rubocop:disable Style/DateTime
```

Comment:
We could be RuboCop friendly by changing this line to:
```suggestion
          Timecop.freeze(tomorrow) do
```

Then, we could add the tomorrow method:

```ruby
def tomorrow
  Time.now + 86400
end
```